workos/resources/directory_sync.py

```diff
@@ -92,3 +97,12 @@ def to_dict(self):
         directory_group = super(WorkOSDirectoryUser, self).to_dict()
 
         return directory_group
+
+    def primary_email(self, user):
+        primary_email_array = []
```

Comment:
I think we could simplify the implementation here using something along the lines of a `.find` (which seems to be curiously absent from Python):

```py
def primary_email(self):
    return next(email for email in user["emails"] if email["primary"], None)
```

Here we're using a list comprehension to filter down the emails to the ones where `primary` is truthy. We then use [`next`](https://docs.python.org/3.7/library/functions.html#next) to return the next item from the iterator. The `None` as the second argument is what we return if there are no primary emails (it will raise a `StopIteration` exception otherwise).

Reply:
Thank you for explaining this, that makes it look much cleaner! 🤌

Reply:
Hey @maxdeviant - I'm having a bit of trouble getting the tests to pass for both Python 2.7 and 3.7. Currently they are only passing for 3.7. 

the Python 2.7 Semaphore run is failing pytest for supplying the primary_email method with a dict instead of an instance of the WorkOSDirectoryUser class. In the SDK, the get_user method returns a dict, not an instance of the class, so it is my assumption that this method should indeed take a dict, not an instance of the class. 

Right now, when I structure the call like the below it returns the primary email as expected.
```primary_email = WorkOSDirectoryUser.primary_email(mock_user_instance)```

Since the SDK returns a dict, calling the method like the below here throws an error regarding a dict not having the method primary_email(). 
```mock_user_instance.primary_email()```

When I adjust the get_user() method to return an instance of the WorkOSDirectoryUser class itself, I get an error referencing the class not being subscriptable when attempting to access the primary_email method. 

I'm not sure how to approach this in order for it to get around this for both Python 2.7 or 3.7 while maintaining the syntax that you had suggested for the method. Do you have any ideas on how to best approach this?

Reply:
If I'm understanding this correctly, it sounds like the issue here is that the `get_user` method is returning a plain dictionary instead of the `WorkOSDirectoryUser` class.

I think for the purposes of this PR we should test the `primary_email` method in isolation based on an instance of the `WorkOSDirectoryUser`.

We'll then eventually want to make `get_user` return a `WorkOSDirectoryUser`. This is a breaking change though, so we'll need to manage that.

Reply:
Sounds good, I got the tests working now! 👍🏽